File: ruby/gmsec4/dox/SchemaIDIterator.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.


module Libgmsec_ruby #:nodoc: don't document this

	##
	# This class supports the iteration of a Specification object's loaded schema (message) templates.
	#
	# ==== See also
	#
	# Libgmsec_ruby::Specification.get_schema_id_iterator
	#
	class SchemaIDIterator
		##
		# Reports whether there are additional schema IDs that can be referenced.
		#
		# ==== Returns
		#
		# Returns true if additional schema IDs are available; false otherwise.
		#
		# ==== See also
		#
		# SchemaIDIterator.next
		#
		def has_next
		end


		##
		# Returns the next available schema ID.
		#
		# ==== Returns
		#
		# A schema ID string.
		#
		# ==== Raises
		#
		# A GmsecException is thrown if the iterator has reached the end (i.e. there are no more schema IDs).
		#
		# ==== See also
		#
		# SchemaIDIterator.has_next
		#
		def next
		end


		##
		# Resets the iterator to the beginning of the schema list that is maintained by Specification.
		#
		def reset
		end
	end
end

File: nodejs/gmsec5/dox/SchemaIDIterator.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file SchemaIDIterator.java
 */

package gmsec;


/**
 * Supports the iteration over the list of schema templates maintained by a Specification object.
 *
 * @see Specification
 */
public class SchemaIDIterator
{
	/**
	 * Indicates whether there are additional Schema IDs that can be referenced using next().
	 *
	 * @return True is returned if additional fields are available, false otherwise.
	 */
	public hasNext();


	/**
	 * Returns the next available Schema ID.
	 *
	 * @return A Schema ID string.
	 *
	 * @throws GmsecException Thrown if the iterator has reached the end (i.e. there are no more IDs).
	 */
	public next();


	/**
	 * Resets the iterator to the beginning of the schema list that is maintained by the Specification object.
	 */
	public reset();
}
